endpoints/setuid_test.go

```diff
@@ -18,67 +19,163 @@ import (
 	metricsConf "github.com/prebid/prebid-server/pbsmetrics/config"
 )
 
-func TestNormalSet(t *testing.T) {
-	response := doRequest(makeRequest("/setuid?bidder=pubmatic&uid=123", nil), true, false)
-	assertIntsMatch(t, http.StatusOK, response.Code)
-	assertHasSyncs(t, response, map[string]string{
-		"pubmatic": "123",
-	})
-}
-
-func TestUnset(t *testing.T) {
-	response := doRequest(makeRequest("/setuid?bidder=pubmatic", map[string]string{"pubmatic": "1234"}), true, false)
-	assertIntsMatch(t, http.StatusOK, response.Code)
-	assertHasSyncs(t, response, nil)
-}
-
-func TestMergeSet(t *testing.T) {
-	response := doRequest(makeRequest("/setuid?bidder=pubmatic&uid=123", map[string]string{"rubicon": "def"}), true, false)
-	assertIntsMatch(t, http.StatusOK, response.Code)
-	assertHasSyncs(t, response, map[string]string{
-		"pubmatic": "123",
-		"rubicon":  "def",
-	})
-}
-
-func TestGDPRPrevention(t *testing.T) {
-	response := doRequest(makeRequest("/setuid?bidder=pubmatic&uid=123", nil), false, false)
-	assertIntsMatch(t, http.StatusOK, response.Code)
-	assertStringsMatch(t, "The gdpr_consent string prevents cookies from being saved", response.Body.String())
-	assertNoCookie(t, response)
-}
-
-func TestGDPRConsentError(t *testing.T) {
-	response := doRequest(makeRequest("/setuid?bidder=pubmatic&uid=123&gdpr_consent=BONciguONcjGKADACHENAOLS1rAHDAFAAEAASABQAMwAeACEAFw", nil), false, true)
-	assertIntsMatch(t, http.StatusBadRequest, response.Code)
-	assertStringsMatch(t, "No global vendor list was available to interpret this consent string. If this is a new, valid version, it should become available soon.", response.Body.String())
-	assertNoCookie(t, response)
-}
-
-func TestInapplicableGDPR(t *testing.T) {
-	response := doRequest(makeRequest("/setuid?bidder=pubmatic&uid=123&gdpr=0", nil), false, false)
-	assertIntsMatch(t, http.StatusOK, response.Code)
-	assertHasSyncs(t, response, map[string]string{
-		"pubmatic": "123",
-	})
-}
+func TestSetUIDEndpoint(t *testing.T) {
+	testCases := []struct {
+		uri                   string
+		existingSyncs         map[string]string
+		gdprAllowsHostCookies bool
+		gdprReturnsError      bool
+		expectedSyncs         map[string]string
+		expectedRespMessage   string
+		expectedResponseCode  int
+		description           string
+	}{
+		{
+			uri:                   "/setuid?bidder=pubmatic&uid=123",
```

Comment:
Is it common in Go to write mini test runners versus breaking them out into individual tests?

Reply:
Talked offline. This is a common design pattern for tests in Go. See:
- https://github.com/golang/go/wiki/TableDrivenTests
- https://dave.cheney.net/2019/05/07/prefer-table-driven-tests